rsconf/component/__init__.py

```diff
@@ -332,6 +329,38 @@ def _bash_append_and_dst(
         self._bash_append(host_path, md5=md5)
         return dst
 
+    def _gen_host_and_identity_ssh_keys(
```

Comment:
Seems like this should be public.

Reply:
I get tripped up in Python with classes and what should be public but I believe since only instances of the class call this method (using self or super) private is correct. Agree?

```python
rsconf (210-sirepo-sandbox)$ gpy _gen_host_and_identity_ssh_keys
./rsconf/component/__init__.py:332:    def _gen_host_and_identity_ssh_keys(
./rsconf/component/devbox.py:34:        z.secrets = self._gen_host_and_identity_ssh_keys(jc)
./rsconf/component/devbox.py:85:    def _gen_host_and_identity_ssh_keys(self, jc):
./rsconf/component/devbox.py:89:        res = super()._gen_host_and_identity_ssh_keys(
./rsconf/component/mpi_worker.py:26:        z.secrets = self._gen_host_and_identity_ssh_keys(jc, z)
./rsconf/component/mpi_worker.py:95:    def _gen_host_and_identity_ssh_keys(self, jc, z):
./rsconf/component/mpi_worker.py:96:        return super()._gen_host_and_identity_ssh_keys(
./rsconf/component/mpi_worker.py:130:            s = self._gen_host_and_identity_ssh_keys(jc, z)
```

Reply:
It's inconsistent with the other methods in component.T. I understand the principle of most private. The only guidance in PEP8 is: 

> Use one leading underscore only for non-public methods and instance variables.

and

> Another category of attributes are those that are part of the “subclass API” (often called “protected” in other languages). Some classes are designed to be inherited from, either to extend or modify aspects of the class’s behavior. When designing such a class, take care to make explicit decisions about which attributes are public, which are part of the subclass API, and which are truly only to be used by your base class.

Underscores do not seem to come into play here other than reading between the lines.

I think there is no such thing as "protected" in Python. In BOP we labeled methods with an "internal_" prefix to mean protected, but Perl adopts no such convention and there is no name mangling like in Python for double underscore.

In this particular case, I vote for consistency for the other functions. You certainly can add this as a /discuss issue.

Reply:
Make sense. Thanks for the details. 367ae5c